net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -834,189 +914,230 @@ mwan3_set_policies_iptables()
+			return
+		fi
+	fi
 
 	[ -z "$dest_ip" ] && unset dest_ip
 	[ -z "$src_ip" ] && unset src_ip
 	[ -z "$ipset" ] && unset ipset
 	[ -z "$src_port" ]  && unset src_port
 	[ -z "$dest_port" ]  && unset dest_port
-	[ "$proto"  != 'tcp' ]  && [ "$proto" != 'udp' ] && {
+	if [ "$proto"  != 'tcp' ]  && [ "$proto" != 'udp' ]; then
 		[ -n "$src_port" ] && {
-			$LOG warn "src_port set to '$src_port' but proto set to '$proto' not tcp or udp. src_port will be ignored"
+			LOG warn "src_port set to '$src_port' but proto set to '$proto' not tcp or udp. src_port will be ignored"
 		}
+
 		[ -n "$dest_port" ] && {
-			$LOG warn "dest_port set to '$dest_port' but proto set to '$proto' not tcp or udp. dest_port will be ignored"
+			LOG warn "dest_port set to '$dest_port' but proto set to '$proto' not tcp or udp. dest_port will be ignored"
 		}
 		unset src_port
 		unset dest_port
-	}
-
-	config_get rule_logging "$1" logging 0
-	config_get global_logging globals logging 0
-	config_get loglevel globals loglevel notice
+	fi
 
 	if [ "$1" != "$(echo "$1" | cut -c1-15)" ]; then
-		$LOG warn "Rule $1 exceeds max of 15 chars. Not setting rule" && return 0
+		LOG warn "Rule $1 exceeds max of 15 chars. Not setting rule" && return 0
 	fi
 
 	if [ -n "$ipset" ]; then
 		ipset="-m set --match-set $ipset dst"
 	fi
 
-	if [ -n "$use_policy" ]; then
-		if [ "$use_policy" = "default" ]; then
-			policy="MARK --set-xmark $MMX_DEFAULT/$MMX_MASK"
-		elif [ "$use_policy" = "unreachable" ]; then
-			policy="MARK --set-xmark $MMX_UNREACHABLE/$MMX_MASK"
-		elif [ "$use_policy" = "blackhole" ]; then
-			policy="MARK --set-xmark $MMX_BLACKHOLE/$MMX_MASK"
-		else
-			if [ "$sticky" -eq 1 ]; then
+	if [ -z "$use_policy" ]; then
+		return
+	fi
 
-				policy="mwan3_policy_$use_policy"
+	if [ "$use_policy" = "default" ]; then
+		policy="MARK --set-xmark $MMX_DEFAULT/$MMX_MASK"
+	elif [ "$use_policy" = "unreachable" ]; then
+		policy="MARK --set-xmark $MMX_UNREACHABLE/$MMX_MASK"
+	elif [ "$use_policy" = "blackhole" ]; then
+		policy="MARK --set-xmark $MMX_BLACKHOLE/$MMX_MASK"
+	else
+		rule_policy=1
+		policy="mwan3_policy_$use_policy"
+		if [ "$sticky" -eq 1 ]; then
+			$IPS -! create "mwan3_sticky_v4_$rule" \
+			     hash:ip,mark markmask "$MMX_MASK" \
+			     timeout "$timeout"
+			[ $NO_IPV6 -eq 0 ] &&
+				$IPS -! create "mwan3_sticky_v6_$rule" \
+				     hash:ip,mark markmask "$MMX_MASK" \
+				     timeout "$timeout" family inet6
+			$IPS -! create "mwan3_sticky_$rule" list:set
+			$IPS -! add "mwan3_sticky_$rule" "mwan3_sticky_v4_$rule"
+			[ $NO_IPV6 -eq 0 ] &&
+				$IPS -! add "mwan3_sticky_$rule" "mwan3_sticky_v6_$rule"
+		fi
+	fi
 
-				for IPT in "$IPT4" "$IPT6"; do
-					[ "$IPT" = "$IPT6" ] && [ $NO_IPV6 -ne 0 ] && continue
-					if ! $IPT -S "$policy" &> /dev/null; then
-						$IPT -N "$policy"
-					fi
+	[ "$ipv" = ipv6 ] && [ $NO_IPV6 -ne 0 ] && return
+	[ "$family" = "ipv4" ] && [ "$ipv" = "ipv6" ] && return
+	[ "$family" = "ipv6" ] && [ "$ipv" = "ipv4" ] && return
```

Comment:
what is `ipv` here? unset?

Reply:
sorry I misunderstood here

Reply:
@ptpt52 You got that right. I don't understand it either at the first place.
I can't find the variable `$ipv`.
But it is a global Variable if so we should capitalize them but I do not like this.
I think the best is to set it as an argument to the function call.
@aaronjg Could we please pass this variable as an function argument.
So we have
`config_foreach mwan3_set_user_iptables_rule rule "$ipv"`
And this is the first argument of the function
```
mwan3_set_user_iptables_rule() {
local ipv="$1"
```

Also you forgot the quotation marks in `ipv6` check in this line with `$ipv`
so we have `[ "$ipv" = "ipv6" ] && [ $NO_IPV6 -ne 0 ] && return`